src/kibana/components/vislib/components/color/color.js

```diff
@@ -23,11 +24,15 @@ define(function (require) {
       });
 
       var arrayLength = arrayOfStringsOrNumbers.length;
-      var colorObj = _.zipObject(arrayOfStringsOrNumbers, createColorPalette(arrayLength));
+      var colors = createColorPalette(arrayLength + mappedColors.count());
+      var uniqueColors = _.difference(colors, mappedColors.all()).slice(0, arrayLength + 1);
+      var colorObj = _.zipObject(arrayOfStringsOrNumbers, uniqueColors);
 
       return function (value) {
-        return colorObj[value];
+        if (!mappedColors.get(value))
+          mappedColors.add(value, colorObj[value]);
```

Comment:
conditional statements over two lines should include brackets: `{}`.
For example:

```javascript
if (!mappedColors.get(value)) {
  mappedColors.add(value, colorObj[value]);
}
```